src/ClientData/ScopeIdProvider.cpp

```diff
@@ -26,10 +30,20 @@ namespace orbit_client_data {
                 std::begin(instrumented_functions), std::end(instrumented_functions),
                 [](const auto& a, const auto& b) { return a.function_id() < b.function_id(); })
                 ->function_id();
-  return std::unique_ptr<NameEqualityScopeIdProvider>(new NameEqualityScopeIdProvider(max_id + 1));
+
+  absl::flat_hash_map<uint64_t, std::string> scope_id_to_name;
+  std::transform(std::begin(instrumented_functions), std::end(instrumented_functions),
+                 std::inserter(scope_id_to_name, std::end(scope_id_to_name)),
+                 [](const auto& instrumented_function) {
+                   return std::make_pair(instrumented_function.function_id(),
+                                         llvm::demangle(instrumented_function.function_name()));
```

Comment:
There should be no need to demangle the function. By now, we only should have demangled names.